File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/template/ProjectTemplateAdminService.java
package ma.zs.zyn.service.facade.admin.template;

import java.util.List;
import ma.zs.zyn.bean.core.template.ProjectTemplate;
import ma.zs.zyn.dao.criteria.core.template.ProjectTemplateCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface ProjectTemplateAdminService extends  IService<ProjectTemplate,ProjectTemplateCriteria>  {

    List<ProjectTemplate> findByCategoryProjectTemplateId(Long id);
    int deleteByCategoryProjectTemplateId(Long id);
    long countByCategoryProjectTemplateId(Long id);
    List<ProjectTemplate> findBySubDomainTemplateId(Long id);
    int deleteBySubDomainTemplateId(Long id);
    long countBySubDomainTemplateId(Long id);




}

File: frontend/src/app/controller/service/admin/template/ProjectTemplateAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {ProjectTemplateDto} from 'src/app/controller/model/template/ProjectTemplate.model';
import {ProjectTemplateCriteria} from 'src/app/controller/criteria/template/ProjectTemplateCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class ProjectTemplateAdminService extends AbstractService<ProjectTemplateDto, ProjectTemplateCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/projectTemplate/');
    }

    public constrcutDto(): ProjectTemplateDto {
        return new ProjectTemplateDto();
    }

    public constrcutCriteria(): ProjectTemplateCriteria {
        return new ProjectTemplateCriteria();
    }
}
